src/backend/Dockerfile

```diff
@@ -22,19 +22,43 @@ RUN set -ex \
         "libgdal-dev" \
     && rm -rf /var/lib/apt/lists/*
 
-# Install PDM
-RUN pip install -U pdm
-# Disable update check
-ENV PDM_CHECK_UPDATE=false
-
 WORKDIR /project
 
+# Install uv
+COPY --from=ghcr.io/astral-sh/uv:0.5.2 /uv /bin/uv
+
+# Ensure executables are at the front of the path
+ENV PATH="/bin:$PATH"
+
+# - Silence uv complaining about not being able to use hard links,
+# - tell uv to byte-compile packages for faster application startups,
+# - prevent uv from accidentally downloading isolated Python builds,
+# - use a temp dir instead of cache during install,
+# - select system python version,
+# - declare `/opt/python` as the target for `uv sync` (i.e. instead of .venv).
+ENV LANG=en_US.UTF-8 \
+    LANGUAGE=en_US:en \
+    LC_ALL=en_US.UTF-8 \
+    UV_LINK_MODE=copy \
+    UV_COMPILE_BYTECODE=1 \
+    UV_PYTHON_DOWNLOADS=never \
+    UV_NO_CACHE=1 \
+    UV_PYTHON="python$PYTHON_BASE" \
+    UV_PROJECT_ENVIRONMENT=/opt/python
+STOPSIGNAL SIGINT
+
 # Copy files needed for building the project
-COPY ./pyproject.toml /project/
-COPY ./pdm.lock /project/
+COPY ./pyproject.toml ./alembic.ini ./uv.lock /project/
+
+# Install dependencies using uv (caching to improve speed for repeated builds)
+RUN --mount=type=cache,target=/root/.cache/uv \
+    --mount=type=bind,source=pyproject.toml,target=pyproject.toml \
+    --mount=type=bind,source=uv.lock,target=uv.lock \
+    uv sync --frozen --no-install-project
 
-# Install dependencies and project into the local packages directory
-RUN pdm install --check --prod --no-editable
+# Sync the project again using uv, ensuring all project dependencies are installed
+RUN --mount=type=cache,target=/root/.cache/uv \
```

Comment:
What's the reason for the double `sync`?

Reply:
> What's the reason for the double `sync`?

```# Install dependencies using uv (caching to improve speed for repeated builds)
RUN --mount=type=cache,target=/root/.cache/uv \
    --mount=type=bind,source=pyproject.toml,target=pyproject.toml \
    --mount=type=bind,source=uv.lock,target=uv.lock \
    uv sync --frozen --no-install-project

# Sync the project again using uv, ensuring all project dependencies are installed
RUN --mount=type=cache,target=/root/.cache/uv \
    uv sync
```
I found, the first uv sync with --no-install-project caches dependencies, while the second installs the project after copying the code. This speeds up builds. Details: [[link](https://docs.astral.sh/uv/guides/integration/docker/#intermediate-layers)].